pkg/localConfig/localConfig.go

```diff
@@ -58,12 +57,12 @@ func (lc *LocalConfigClient) GetLocalConfiguration() (*LocalConfig, error) {
 	offline := viper.GetString(offlineKey)
 
 	if offline == "" {
-		viper.SetDefault(offlineKey, "fail")
-		_ = viper.WriteConfig()
-		_ = viper.ReadInConfig()
-		offline = viper.GetString(offlineKey)
+		offline = "fail"
+		err := setViperVariable(offlineKey, offline)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Let's make sure we don't need to `return &LocalConfig{}, err` instead of `return nil, err`